packages/embark/templates/demo/app/components/whisper.js

```diff
@@ -45,17 +45,16 @@ class Whisper extends React.Component {
       subscribedChannels
     });
 
-    EmbarkJS.Messages.listenTo({topic: [this.state.listenTo]}, (error, message) => {
-      const messageList = this.state.messageList;
-      if (error) {
-        messageList.push(<span className="alert-danger">Error: {error}</span>);
-      } else {
+    const messageList = this.state.messageList;
+    const action = (fn) => (...args) => { fn(...args); this.setState({messageList}); };
```

Comment:
This line is very complex, especially for a new user. Since this is for the demo, can you use an easier syntax as before or put comments, please.

Reply:
How about this?

```js
EmbarkJS.Messages.listenTo({topic: [this.state.listenTo]}).subscribe(
      message => {
        messageList.push(<span>Channel: <b>{message.topic}</b> |  Message: <b>{message.data}</b></span>);
        this.setState({messageList});
      },
      error => {
        messageList.push(<span className="alert-danger">Error: {error}</span>);
        this.setState({messageList});
      }
);
```
It's less DRY but easier to follow, for sure.

Reply:
That's great!

Reply:
Done. See: 849a8a02fb0712bcf0ddb3d02e2a5026adff67fb.

Reply:
Much better. 👌